timetracker/entries/models.py

```diff
@@ -2,13 +2,51 @@
 
 from django.utils import timezone
 
+class Country(models.Model):
+	name = models.CharField(max_length=100, null=True)
+	
+	class Meta:
+		verbose_name_plural = "Countries"
+	
+	def __str__(self):
+		return self.name
+
+class Address(models.Model):
+	unit_no = models.CharField(max_length=15, default=None, blank=True, null=True)
+	street_no = models.CharField(max_length=20, default=None, blank=True, null=True)
+	street_name = models.CharField(max_length=100, null=True)
+	suburb = models.CharField(max_length=50, null=True)
+	state = models.CharField(max_length=50, null=True)
+	country = models.ForeignKey('Country', null=True)
+	
+	
+	class Meta:
+		verbose_name_plural = "Addresses"
+	
+	def __str__(self):
+		return "{unit}{st_no}{st_name}, {suburb}, {state}, {country}".format(
+			unit="Unit {}/".format(self.unit_no) if self.unit_no else "",
+			st_no = "{} ".format(self.street_no) if self.street_no else "",
+			st_name=self.street_name,
+			suburb=self.suburb,
+			state=self.state,
+			country=self.country)
+
+
+class Client(models.Model):
+	business_name = models.CharField(max_length=200, null=True)
+	address = models.ForeignKey('Address', null=True)
+	
+	def __str__(self):
+		return '{} of {}'.format(self.business_name, self.address)
+
 
 class Project(models.Model):
-    client = models.CharField(max_length=200)
-    name = models.CharField(max_length=200)
+    client = models.ForeignKey('Client', null=True)
```

Comment:
This shouldn't be nullable, as we want _all_ `Projects` to have a `Client`.

The way to avoid this being a problem when adding a new `ForeignKey` is to add the field nullable, do a data migration to set values, then remove the `null=True` property.